### ISSUE TITLE
elastic: pretty_select does not display popover if popover would overlap <select> element

### ISSUE THREAD

**Issue by bbraybrook** [roundcube/roundcubemail#6769]
if the popover to be displayed by `pretty_select` is large enough that it would obscure the `<select>` element it's bound to, bootstrap.popover will not display the popover (due to the way Popper.js deals with the "flip" option).

it's easy to trigger this on `settings -> preferences -> special folders` when in mobile/small view with a large folder list or with the select element being clicked on near the bottom of the screen.

the `show.bs.popover` event fires but the `shown.bs.popover` does not.

this only affects the automatic showing of the popover via the click handler; calling `$(select_elem).popover.show()` once the element does not display will display the popover overtop the `<select>` element. calling `$(select_elem).popover.show()` on a popover that is already displayed has no effect - the popover remains in place.

a proposed fix, using `setTimeOut` to detect when a desired popover does not display. perhaps there is a more clever way to do this, or the variable used should be inside `rcube_elastic_ui` instead:

```
--- ui.js	2019-05-23 18:10:42.000000000 +0000
+++ ui.js.new	2019-05-23 18:17:02.000000000 +0000
@@ -3518,6 +3518,7 @@
                         + '<div class="popover-body" style="max-height: ' + max_height + 'px"></div></div>'
                 })
                 .on('shown.bs.popover', function() {
+                    if (rcmail.popover_flip_sanity) delete rcmail.popover_sanity; // // clear sanity check for popover not displaying 
                     select.focus(); // for Chrome
                     // Set popup Close title
                     list.parent().prev()
@@ -3537,6 +3538,18 @@
                     // don't propagate mousedown event
                     list.on('mousedown', function(e) { e.stopPropagation(); });
                 })
+                .on('show.bs.popover', function() {
+                    // on small screens, if the popover is set to overlap the parent element the
+                    // popover will not display. we can trigger it by forcibly calling show.
+                    rcmail.popover_flip_sanity = this;
+                    setTimeout(function() {
+                    if (rcmail.popover_flip_sanity) {
+                        // If the client is slow and this gets called when the popover has already
+                        // been shown, the show call does nothing and no harm is done
+                        $(rcmail.popover_flip_sanity).popover('show');
+                        delete rcmail.popover_flip_sanity;
+                    }
+                },100);
                 .popover('show');
         };
```

**Comment by zjyysc11** [comment]
什么

**Comment by alecpl** [comment]
I'm unable to reproduce. Any specific browser? Are you using git-master?

**Comment by bbraybrook** [comment]
sorry, forgot to mention those details.

this was with chrome, and roundcube-1.4.rc1. firefox seems fine.

it does seem to be fixed in master, though (i just dropped master's ui.js in and despite some issues with the menus/list not getting hidden, the dropdowns did work properly once i manually hid the other stuff) - some other change must have fixed the problem.

sorry for the false alarm on the bug.

**Comment by alecpl** [comment]
Probably this is the fix: #6705.
